### DISCUSSION TITLE
Access the underlying NSTextView?

### DISCUSSION THREAD

**Discussion by S1D1T1** [danielsaidi/RichTextKit#165] [Q&A]
Is there a preferred way to access the NSTextView underlying a RichTextEditor?

I came across this tip for setting the "placeholder text" for a field. While NSTextView doesn't have a function to set it, you can access it with the generic `setValue( forKey:)`. The only way I found for applying that to the NSTextView was in the viewConfiguration closure which I pass to `RichTextEditor()`. I cast its argument as a NSTextView. Likewise, I set other flags in that closure as well. But this doesn't give much programmatic access - if you wanted to set these things later. 

```swift
		viewConfiguration:  {
	if let textView = $0 as? NSTextView{
		textView.setValue(isNeg ? " What NOT to include": " What to include in your picture",
				 			forKey: "placeholderString")
		textView.isAutomaticSpellingCorrectionEnabled = false
		textView.isContinuousSpellCheckingEnabled = false 
	}
}```

**Comment by danielsaidi** [comment]
Hi @S1D1T1 

I missed this discussion, sorry about that!

Ideally, the `RichTextViewComponent` should expose everything you need, but there sure are a lot of functionality that will be hard to expose through this protocol. At the same time, exposing the underlying types kind of ruins the platform-agnostic approach, so I'm not sure how to best do it.

In the case of the placeholder, that would be a great addition to the component protocol.

**Reply by S1D1T1** [reply]
agreed that an optional `placeholder:String` parameter to `RichTechEditor()` would be appropriate - I originally looked for the option when installing. this modification has worked perfectly with my MacOS app.

<img width="638" alt="Screenshot 2024-05-03 at 8 11 10 AM" src="https://github.com/danielsaidi/RichTextKit/assets/156350598/f7eb372d-75ee-42b4-ae5f-f37b69f48179">
